runtime/src/manifest.rs

```diff
@@ -45,10 +45,7 @@ fn to_module(engine: &Engine, wasm: &extism_manifest::Wasm) -> Result<(String, M
 
             // Figure out a good name for the file
             let name = match &meta.name {
-                None => {
-                    let name = path.with_extension("");
-                    name.file_name().unwrap().to_string_lossy().to_string()
-                }
+                None => meta.name.as_deref().unwrap_or(MAIN_KEY).to_string(),
```

Comment:
Do we want to keep track of the names we've seen so we can error on duplicates? (It's possible we'll use the default `MAIN_KEY` multiple times if there are multiple unnamed modules.)

Reply:
Thanks, just added a check for duplicate module names!